Error looking for test_dir on python 3.8

Thanks for sharing this plugin, it's very interesting.

I gave it a try, but:
```
install installed: attrs==19.3.0,jmespath==0.10.0,more-itertools==8.3.0,packaging==20.4,pluggy==0.13.1,portalocker==1.7.0,py==1.8.1,pyparsing==2.4.7,pytest==5.4.3,pytest-terraform @ git+https://github.com/cloud-custodian/pytest-terraform@caa1585a0ed996c7bc17629b1262b54d77e45bd7,six==1.15.0,wcwidth==0.2.3
install run-test-pre: PYTHONHASHSEED='1943363608'
install run-test: commands[0] | pytest install
============================================================ test session starts ============================================================
platform linux -- Python 3.8.3, pytest-5.4.3, py-1.8.1, pluggy-0.13.1
cachedir: /home/foo/bar/tests/.tox/install/.pytest_cache
rootdir: /home/foo/bar/tests/install, inifile: pytest.ini
plugins: terraform-0.1.0
collected 0 items / 1 error                                                                                                                 

================================================================== ERRORS ===================================================================
_____________________________________________________ ERROR collecting test_install.py ______________________________________________________
install/test_install.py:4: in <module>
    class TestAnsible:
install/test_install.py:6: in TestAnsible
    @terraform('openstask', scope='session')
.tox/install/lib/python3.8/site-packages/pytest_terraform/tf.py:333: in __call__
    test_dir = local(f.f_locals["__file__"]).dirpath()
E   KeyError: '__file__'
========================================================== short test summary info ==========================================================
ERROR install/test_install.py - KeyError: '__file__'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================= 1 error in 0.08s ==============================================================
```

On archlinux, installed via a tox file:
```
[testenv:install]
skip_install = true
deps =
   pytest
   git+https://github.com/cloud-custodian/pytest-terraform
   jmespath
   portalocker
commands = pytest install {posargs}
```

and a super-simple test-file 

```
from pytest_terraform import terraform

class TestAnsible:

    @terraform('openstack', scope='session')
    def test_install(openstack):
        print(openstack)
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "pytest-terraform"
-version = "0.1.0"
+version = "0.1.1"
 description = "A pytest plugin for using terraform fixtures"
 authors = ["Kapil Thangavelu <kapilt@gmail.com>"]
 license = "Apache-2.0"

--- a/pytest_terraform/tf.py
+++ b/pytest_terraform/tf.py
@@ -330,7 +330,7 @@ class FixtureDecoratorFactory(object):
         # on the other.
         f = sys._getframe(1)
         name = name or terraform_dir
-        test_dir = local(f.f_locals["__file__"]).dirpath()
+        test_dir = local(_frame_path(f)).dirpath()
         if replay is None:
             replay = LazyReplay.resolve()
         found = None
@@ -354,6 +354,15 @@ class FixtureDecoratorFactory(object):
     @staticmethod
     def nonce_decorator(func):
         return func
+
+
+def _frame_path(f):
+    start = f
+    while f:
+        if "__file__" in f.f_locals:
+            return f.f_locals["__file__"]
+        f = f.f_back
+    raise RuntimeError("frame path not found %s" % start)
 
 
 terraform = FixtureDecoratorFactory()

--- a/tests/test_terraform.py
+++ b/tests/test_terraform.py
@@ -5,6 +5,20 @@ import pytest
 from pytest_terraform import tf
 
 
+def test_frame_walk():
+    class Frame:
+        def __init__(self, f_locals, f_back=None):
+            self.f_locals = f_locals
+            self.f_back = f_back
+
+    parent = Frame({"__file__": "/tmp/file.py"})
+    child = Frame({}, parent)
+    assert tf._frame_path(child) == "/tmp/file.py"
+
+    with pytest.raises(RuntimeError):
+        tf._frame_path(Frame({}))
+
+
 @tf.terraform("local_bar", scope="session")
 def test_tf_user_a(local_bar):
     print("test invoked a")

